### PR TITLE
Adds random map generation to Sif. [IDB IGNORE] [MDB IGNORE] 

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Depends on #8222.
 
- Adds random map generation to the wilderness and plains to replace the current fixed turf flora generation.
- Adds random map generation to the underground to reskin some areas to mud and dirt, and to spawn mushrooms and glowplants.

This significantly increases the initialization delay for map (+50 seconds on my machine).

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Wilderness map:
image
image

Plains map:
image
image

File: polaris.dme [hunk 1]
```diff
@@ -5,6 +5,198 @@
 // END_INTERNALS
 // BEGIN_FILE_DIR
 #define FILE_DIR .
+#define FILE_DIR "code"
```
Comment (MistakeNot4892) [inline polaris.dme:8]: why though

**Comment by MistakeNot4892** [comment]
Splitting this into two PRs, this one now depends on #8222.

**Comment by MistakeNot4892** [comment]
I slipped and this has turned into a wide scale refactor project.

**Comment by MistakeNot4892** [comment]
Will revisit this after #8299 and #8298 are merged.

**Comment by MistakeNot4892** [comment]
Alright, PR has been pruned back to just the map generation and some turf refactoring to allow for mapgen to override grass/tree gen.

This PR takes initialization time fro 95 seconds to 140 seconds so I am not sure if it is actually desirable.

**Comment by Atermonera** [comment]
Map test concluded. Added some time to init, but imo init time is cheap. Usage by downstreams largely requires that they map in these masking turfs.

**Review by Atermonera — APPROVED** [review]

**Comment by Atermonera** [comment]
Staff :+1:

**Pull request merged**